go/host/storage/storage.go

```diff
@@ -110,6 +111,27 @@ func (s *storageImpl) AddBlock(b *types.Header) error {
 	return nil
 }
 
+func (s *storageImpl) FetchCrossChainProof(messageType string, crossChainMessage gethcommon.Hash) ([][]byte, gethcommon.Hash, error) {
+	tree, err := hostdb.GetCrossChainMessagesTree(s.db, crossChainMessage)
+	if err != nil {
+		return nil, gethcommon.Hash{}, err
+	}
+
+	for k, value := range tree {
```

Comment:
isn't this block below reusable?
I have a feeling it's implemented in slightly different ways in the enclave and here.
I'd make it reusable in common

Reply:
Not exactly, the data types from which the tree is built are different so before each smt.of we have to convert it